### PR TITLE
[release-1.6] 📖 Sync version and release docs with main

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Includes changes from #9971 #9972

### CODE REVIEW

**Review by chrischdi — COMMENTED** [review]
/lgtm

**Comment by fabriziopandini** [comment]
> Includes changes from https://github.com/kubernetes-sigs/cluster-api/pull/9971 and a few other PRs

Could you kindly list those PRs for reference?

**Comment by sbueringer** [comment]
> > Includes changes from #9971 and a few other PRs
> 
> Could you kindly list those PRs for reference?

Okay I look them up

**Comment by sbueringer** [comment]
@fabriziopandini List of PRs should be complete now

File: docs/book/src/reference/versions.md [hunk 1]
```diff
@@ -85,7 +87,7 @@ These diagrams show the relationships between components in a Cluster API releas
 | Kubernetes v1.26  | ✓                    | ✓                 | ✓                 | ✓                 |
 | Kubernetes v1.27  |                      | ✓                 | ✓                 | ✓                 |
 | Kubernetes v1.28  |                      |                   | ✓                 | ✓                 |
-
+| Kubernetes v1.29  |                      |                   |                   | ✓ >= v1.6.1       |
```
Comment (fabriziopandini) [inline docs/book/src/reference/versions.md:90]: q: is there a reason why are not cherry picking similar changes for 1.27 (>= v1.4.2) and 1.28 (>= v1.5.1)?

Comment (sbueringer) [reply docs/book/src/reference/versions.md:90]: No

Comment (sbueringer) [reply docs/book/src/reference/versions.md:90]: Should be fixed

**Pull request merged**